src/models/hobbiesModel.ts

```diff
@@ -0,0 +1,24 @@
+import { hobbies } from "../data/hobbies.js";
+
+type updateHobbyData = {
+  hobbies: string[];
+}
+export const getUserHobbies = (id: string) => {
+  return new Promise((resolve, _reject) => {
+    let userHobbies = hobbies.find((item) => item.id === id);
+    if (!userHobbies) {
+      hobbies.push({ id, hobbies: [] });
+      resolve({ id, hobbies: [] })
+      return;
+    };
+    resolve(userHobbies);
+  });
+};
```

Comment:
```suggestion
export const getUserHobbies = async (id: string) => {
    let userHobbies = hobbies.find((item) => item.id === id);
    if (!userHobbies) {
      hobbies.push({ id, hobbies: [] });
      return { id, hobbies: [] };
    };
   return userHobbies;
  });
};
```
You don't really need promises here.